swarm/storage/mru/doc.go

```diff
@@ -0,0 +1,52 @@
+// Mutable resource is an entity which allows updates to a resource
+// without resorting to ENS on each update.
+// The update scheme is built on swarm chunks with chunk keys following
+// a predictable, versionable pattern.
+//
+// Updates are defined to be periodic in nature, where periods are
+// expressed in terms of number of blocks.
+//
+// The root entry of a mutable resource is tied to a unique identifier,
+// typically - but not necessarily - an ens name.  The identifier must be
+// an valid IDNA string. It also contains the block number
+// when the resource update was first registered, and
+// the block frequency with which the resource will be updated, both of
+// which are stored as little-endian uint64 values in the database (for a
+// total of 16 bytes). It also contains the unique identifier.
+// It is stored in a separate content-addressed chunk (call it the metadata chunk),
+// with the following layout:
+//
+// (0x0000|startblock|frequency|publickey|identifier)
+//
+// (The two first zero-value bytes are used for disambiguation by the chunk validator,
```

Comment:
why does the validator need anything here, this is content addressed.

Reply:
Good point.